### PR TITLE
initial commit BKMD-2148

### PR DESCRIPTION
**Pull request by meirshal**
It looks like BetterDoctor's API doesn't return the NPI of practices, so I couldn't test two of the services. I don't think that this is how it's meant to be so I mailed them and I'm waiting for a reply on that.

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+ * Sends GET requests to BetterDoctor's API
+ * @param url
+ * @param params
+ * @returns {Promise}
+ * @private
+ */
+BetterDoctorApi.prototype._sendRequest = function (url, params) {
+  params = params || {};
+  params.user_key = this.apiAccessKey;
+  Object.keys(params).forEach(function(key) { // convert every parameter that is not a string to a string
```
Comment (chenrozenes) [inline lib/index.js]: I think its nicer to use lodash

File: lib/index.js [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+    if (typeof params[key] != 'string') {
+      params[key] = params[key].toString();
+    }
+  });
+  return request.get({ baseUrl: this.baseUrl, url: url, qs: params }).then(function(result) { return JSON.parse(result); });
```
Comment (chenrozenes) [inline lib/index.js]: Its always more clear to write ```then``` in a new line

File: spec/lib/betterdoctorAPi.spec.js [hunk 3]
```diff
@@ -0,0 +1,154 @@
+/**
+ * Created by meirshalev on 09/02/2017.
+ */
+
+var BetterDoctorApi = require('../../index');
+
+const BASE_URL = 'https://api.betterdoctor.com/2016-03-01';
+const APP_CODE = 'c9024b7115daf4607923e6ade834a744'; //
```
Comment (chenrozenes) [inline spec/lib/betterdoctorAPi.spec.js]: // is redundant

File: lib/index.js [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+  return this._sendRequest(`/practices/${uid}`, params);
+};
+
+/**
+ * Retrieves practice by npi code.
+ * @param npi practice npi code
+ * @param params look at BetterDoctor's API documentation for details https://developer.betterdoctor.com/documentation15#/
+ * @returns {Promise}
+ */
+BetterDoctorApi.prototype.retrievePracticeDescByNapi = function (npi, params) {
```
Comment (chenrozenes) [inline lib/index.js]: I think you meant Npi and not Napi

File: lib/index.js [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+  return this._sendRequest(`/doctors/${uid}`, params);
+};
+
+/**
+ * Retrieves a doctor using npi code.
+ * @param npi doctor npi number
+ * @param params look at BetterDoctor's API documentation for details https://developer.betterdoctor.com/documentation15#/
+ * @returns {Promise}
+ */
+BetterDoctorApi.prototype.retrieveDoctorDescByNapi = function (npi, params){
```
Comment (chenrozenes) [inline lib/index.js]: Same here, I think you mean Npi and not Napi

File: spec/support/jasmine.json [hunk 6]
```diff
@@ -0,0 +1,6 @@
+{
+  "spec_dir": "spec",
```
Comment (chenrozenes) [inline spec/support/jasmine.json]: I like chai more, but thats cool :)

File: spec/lib/betterdoctorAPi.spec.js [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+
+function checkDoctorResult(result) {
+  expect(result.meta.count).toEqual(1);
+  expect(result.data.npi).toEqual(TEST_DOCTOR.npi);
+  expect(result.data.uid).toEqual(TEST_DOCTOR.uid);
+}
+
+describe("BetterDoctorAPI", function() {
+  beforeAll(function() {
+    betterdoctorApi = new BetterDoctorApi(BASE_URL, APP_CODE);
```
Comment (chenrozenes) [inline spec/lib/betterdoctorAPi.spec.js]: You can use the ```this``` context to save variables in ```beforeAll``` and then access them from within the test, so you don't need to declare it before.
Just write ```this.betterdoctorApi``` here, and access it from your tests in the same way.

File: spec/lib/betterdoctorAPi.spec.js [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+  });
+
+  describe('#doctorSearch', function() {
+    it("returns a list of doctors named Levy", function(done) {
+      var name = TEST_DOCTOR.name;
+
+      betterdoctorApi.doctorSearch({ name: name, limit: 1 }).then(function(result) {
```
Comment (chenrozenes) [inline spec/lib/betterdoctorAPi.spec.js]: I like to add ```catch``` to the promise I test, and to provide the ```done``` function in that case as well.
For example:
```
betterdoctorApi.doctorSearch(...).then(function (result) {
...
done()
}).catch(done)
```

File: lib/index.js (hunk 1 above)
Comment (meirshal) [inline lib/index.js]: done

File: lib/index.js (hunk 2 above)
Comment (meirshal) [inline lib/index.js]: fixed

File: lib/index.js (hunk 5 above)
Comment (meirshal) [inline lib/index.js]: apparently old habits die hard...
fixed.

File: lib/index.js (hunk 4 above)
Comment (meirshal) [inline lib/index.js]: fixed

File: spec/lib/betterdoctorAPi.spec.js (hunk 3 above)
Comment (meirshal) [inline spec/lib/betterdoctorAPi.spec.js]: fixed

File: spec/lib/betterdoctorAPi.spec.js (hunk 7 above)
Comment (meirshal) [inline spec/lib/betterdoctorAPi.spec.js]: done

File: spec/lib/betterdoctorAPi.spec.js (hunk 8 above)
Comment (meirshal) [inline spec/lib/betterdoctorAPi.spec.js]: done

**Pull request merged**